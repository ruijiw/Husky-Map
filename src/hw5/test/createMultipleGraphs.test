# Specification testing file.
# Tests that 6 graphs can be created, a 2 nodes for each
# graph an edge between them can be added, and the two nodes
# each graph can be printed


# Create graphs
CreateGraph graph1
CreateGraph graph2
CreateGraph graph3
CreateGraph graph4
CreateGraph graph5
CreateGraph graph6
CreateGraph graph7

# Add 2 nodes to each graph
AddNode graph1 n1
AddNode graph1 n2
AddNode graph2 n1
AddNode graph2 n2
AddNode graph3 n1
AddNode graph3 n2
AddNode graph4 n1
AddNode graph4 n2
AddNode graph5 n1
AddNode graph5 n2
AddNode graph6 n1
AddNode graph6 n2
AddNode graph7 n1
AddNode graph7 n2

# Add an edge to each graph
AddEdge graph1 n1 n2 e1
AddEdge graph2 n1 n2 e1
AddEdge graph3 n1 n2 e1
AddEdge graph4 n1 n2 e1
AddEdge graph5 n1 n2 e1
AddEdge graph6 n1 n2 e1
AddEdge graph7 n1 n2 e1

# Print the nodes in the graph and the outgoing edges from n1
ListNodes graph1
ListNodes graph2
ListNodes graph3
ListNodes graph4
ListNodes graph5
ListNodes graph6
ListNodes graph7
ListChildren graph1 n1
ListChildren graph2 n1
ListChildren graph3 n1
ListChildren graph4 n1
ListChildren graph5 n1
ListChildren graph6 n1
ListChildren graph7 n1