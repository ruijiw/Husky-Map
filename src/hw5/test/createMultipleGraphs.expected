# Specification testing file.
# Tests that 6 graphs can be created, a 2 nodes for each
# graph an edge between them can be added, and the two nodes
# each graph can be printed


# Create graphs
created graph graph1
created graph graph2
created graph graph3
created graph graph4
created graph graph5
created graph graph6
created graph graph7

# Add 2 nodes to each graph
added node n1 to graph1
added node n2 to graph1
added node n1 to graph2
added node n2 to graph2
added node n1 to graph3
added node n2 to graph3
added node n1 to graph4
added node n2 to graph4
added node n1 to graph5
added node n2 to graph5
added node n1 to graph6
added node n2 to graph6
added node n1 to graph7
added node n2 to graph7

# Add an edge to each graph
added edge e1 from n1 to n2 in graph1
added edge e1 from n1 to n2 in graph2
added edge e1 from n1 to n2 in graph3
added edge e1 from n1 to n2 in graph4
added edge e1 from n1 to n2 in graph5
added edge e1 from n1 to n2 in graph6
added edge e1 from n1 to n2 in graph7

# Print the nodes in the graph and the outgoing edges from n1
graph1 contains: n1 n2
graph2 contains: n1 n2
graph3 contains: n1 n2
graph4 contains: n1 n2
graph5 contains: n1 n2
graph6 contains: n1 n2
graph7 contains: n1 n2
the children of n1 in graph1 are: n2(e1)
the children of n1 in graph2 are: n2(e1)
the children of n1 in graph3 are: n2(e1)
the children of n1 in graph4 are: n2(e1)
the children of n1 in graph5 are: n2(e1)
the children of n1 in graph6 are: n2(e1)
the children of n1 in graph7 are: n2(e1)